### DISCUSSION TITLE
Upgrade from sbfspot 3.3.1 to 3.5.0

### DISCUSSION THREAD

**Discussion by pa3fat** [SBFspot/SBFspot#612] [Q&A]
After my issue with SBFspot noticed that i am outdated with my 3.3.1 version. So an upgrade to 3.5.0 seems the minimum.

Looking through the Wiki for installation for there is a specific upgrade for the SBFspot.db
"If you are upgrading, use Update_340_SQLite.sql"
Between which version is this needed, looks like 3.4.0 then but is this still available as intermediate step?

Other things to prepare except for backup of the current db, although empty one will probably not harm as i have all data in PVOutput.org. Backup of crontab and current SBFconfig.cfg as it contains all my reference data.

I also noticed that there is a tool for upgrading beyond 3.5.0. but not yet clear if I have to mark all options or leave them when 3.5.0 is ok.

Thanks for your guidance in advance.

**Comment by pa3fat** [comment]
Following the wiki: https://github.com/SBFspot/SBFspot/wiki/Installation-Linux-SQLite????


After command:
tar -xvf V$sbfspot_version.tar.gz -C SBFspot --strip-components 1

notice it extracts SBFspot-3.8.3 with subs into /SBFspot/multiple files. (Correct?)

I notice following file in /home/pi : V3.8.3.tar.gz

Next step would be cd ~/SBFspot/SBFspot as i don understand why i would want to clone the latest git clone via https://github.com/SBFspot/SBFspot.git.

If i go to cd ~/SBFspot/SBFspot is still see old files. Also in /SBFspot i notice old file but home/pi/SBFspot has todays filestamp
/home/pi/SBFspot/SBFspot has old timestamp.
Is there a step missing in the installation wiki ? Do i need to start with a non exisiting /SBFspot?

make sqlite next.
Compilation started by i did not notice any further unpacking of files.....
I aborted this step for now.

Or is the git clone action doing this????

Please your help.

**Comment by pa3fat** [comment]
dropped /SBFspot and below directories.
created new /SBFspot and did the tar extract again. Followed by the git clone. Now i see new files.....

**Reply by sbf-** [reply]
That makes no sense... the cloning will overwrite the version you untarred with the latest version of the repo.
You untar -OR- you clone, not both

**Comment by pa3fat** [comment]
Think i am on 3.8.3 :)

**Reply by sbf-** [reply]
`SBFspot -?` will tell you, but since you cloned the repo it will be 3.9.7 as defined in [version.h](https://github.com/SBFspot/SBFspot/blob/master/SBFspot/version.h?#L1)

**Reply by sbf-** [reply]
And why do you want 3.8.3 specifically?

**Comment by pa3fat** [comment]
i was fighting with the untar as it was not producing the files in the SBFspot/SBFspot all had a timestamp of i think dec-2021.
Dropped the /SBFspot complete and did the same. Not 100% sure if i checked it was there.

Anywere i can validate if i am actually on 3.9.7?

**Reply by sbf-** [reply]
SBFspot -?

**Reply by sbf-** [reply]
Untar will take the timestamps as stored in the tar. Cloning will create files with current timestamps.

**Reply by pa3fat** [reply]
I am on 3.9.7 so i am good to go :)

Many thanks.
